src/components/CallingCodePicker/CallingCodePicker.tsx

```diff
@@ -59,42 +63,67 @@ const CallingCodePicker: React.FC<ICallingCodePickerProps> = ({
 }) => {
   const [searchValue, setSearchValue] = useState<string>('');
   const [isPickerOpen, setIsPickerOpen] = useState<boolean>(false);
-  const [selectedCountry, setSelectedCountry] = useState<ICountry | undefined>(undefined);
+  const [selectedCountryState, setSelectedCountryState] = useState<ICountry | undefined>(undefined);
   const [countriesData, setCountriesData] = useState<ICountry[]>(countries);
+  const [isSearching, setIsSearching] = useState<boolean>(false);
   const localCountryCode = RNLocalize.getCountry();
-  const alphaCodeWithFallback = selectedValue || localCountryCode;
-
-  const handleCountrySelect = (itemValue: string) => {
-    onValueChange(itemValue);
-    resetPickerState();
-  };
 
-  const resetPickerState = () => {
-    setIsPickerOpen(s => !s);
+  const resetPickerState = (togglePicker?: boolean) => {
+    togglePicker && setIsPickerOpen(state => !state);
     setSearchValue('');
     setCountriesData(countries);
   };
 
+  const handleCountrySelect = (selectedCountry: ICountry) => {
+    setSelectedCountryState(selectedCountry);
+    onValueChange(selectedCountry.callingCode);
```

Comment:
updating the selected country state after a selection will trigger the `findCountry` function and update the UI accordingly.